### PR TITLE
CIP-0001 | Rework to reflect reality

### PR DESCRIPTION
**Pull request by KtorZ**
This is the first step in a broader work aiming at pushing the CIP in the forefront of _every_ Cardano changes. Before addressing current shortcomings of the process, it is necessary to capture what the process _actually is_ at present.

This rework will serve as a base for a retrospective discussion amongst the editors and, shall be proposed to the 
community for approval and review.

---

[see rendered Markdown](https://github.com/cardano-foundation/CIPs/blob/cip-0001-rework/CIP-0001/README.md)

### CODE REVIEW

**Comment by rphair** [comment]
I can't give this a full read & review before the upcoming retrospective meeting but at first glance it looks excellent & I'll catch up with it in the next day.  Recently I realised how badly we needed this, especially with more community postings ([for example](https://forum.cardano.org/t/idea-for-proposing-a-cardano-improvement-proposal/107720)) suggesting an inappropriately loose definition of "CIP" has been going around.  It will be great to refer people to something more rigorous & updated :sunglasses:

**Review by rphair — COMMENTED** [review]
This is a great step forward, and it really is like a complete rewrite... I'd been planning on reading the diff to focus on the changes, but the entire document is "diff" 🤓 so instead I read the old & new documents side by side.

I'm glad the _Motivation_ section defines exactly how the CIP process relates to the highly marketable & overused term "governance".  As long as I'm not misunderstanding this myself, this establishes that the CIP process is there to _inform_ the governance process, _not_ to _achieve_ it.

I will try to remember the type "Standards Track" has been replaced by "Core" (a more intuitive term I think).

I admit that after >1 year I never really understood the `CIP_Flow.png` diagram... and have seen many cases where it seemed we were jumping around independently of it, both in meetings and on GitHub... so I'm happy to see it finally gone.

I can see _Repository organization_ includes the resolution we made, at the recent review meeting, to habitually double-check the CIP status fields ASAP after each PR is posted.

The _Code of Conduct_ is adapted sensibly enough.  I suppose its "Enforcement" section is necessary for liability reasons but it does park the "ownership" of the CIP process fairly well with the Cardano Foundation... this is something to consider whenever claims of impartiality or vested interest are discussed.

### suggestions

The new `PULL_REQUEST_TEMPLATE` is set to preemptively fix some trivial submissions and help start discussions on the Cardano Forum more productively, but:

 - Question: is this in `.github` so it can be included automatically by the GitHub UI when a CIP pull request is created?
 - Suggestion: if so, can it also be linked from `README.md` text, and made visible in this CIP directory somehow (I don't see it in the directory listing on your branch)... so people can work with this material offline and/or on sites outside of GitHub (e.g. Discourse, e.g. the Cardano Forum)?  Perhaps creating a symlink to make this available at the same location of the removed file `cip-template.md`?

Under _Repository organization_ we talked yesterday about adding an item to the list, probably early on (at the time a CIP is first reviewed after PR is posted)

 - If not already there, add a link in the original PR comment pointing to the CIP document in the author's working branch (for ease of discussion & community participation).

The list of current reviewers, with only 4 members, looks great: but won't be so when we have 10 editors.  I wish GitHub would generate responsive table markup... and perhaps it will by the time the editing committee grows to that size 🤣

People (including me, some time ago) wonder about `cips.cardano.org` - how it's different from GitHub, how it's generated.  A lot of this will be reverse engineered by experienced users going back & forth between the two environments... but for nontechnical people, writers, and managers it would also be helpful to have a short but explicit statement: perhaps one general enough that it will still be true even after the pending overhaul of this site.

 - It could well be that this statement belongs in the top level `README.md` rather than in CIP-0001 itself.
 - I see that doc is also being edited in this branch, and has removed the reference to `cips.cardano.org` along with the whole text of the "Sequence Diagram" section, so perhaps it could be added back in under a sub-heading.

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:25]: I'd like to see this paragraph extended. This is one of the most common confusions I see. I'd like it to be really explicit that:
- Acceptance by the CIP editors does not indicate any judgement on the technical merit of a proposal.
    - For this reason, a CIP being merged into the repository as Draft or Proposed means nothing other than that it is procedurally sound.
- The CIP process provides no guarantee or even process for getting a change accepted (into Cardano or otherwise as appropriate for the CIP). It is up to the author to find alternative channels to make this happen.

I'd also like to see a disclaimer like this on the CIP website. Maybe even a banner on Draft CIPs. At the moment stuff looks very official as soon as it's merged and it goes on the website.

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:39]: must? you say "must" in some other contexts, so I would interpret this as saying they are optional, and I'm not sure if that's what you mean

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:104]: what do the brackets mean?

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:144]: Just checking, I think this is a change in the process, suggesting that every merged CIP should be at least in Proposed status, and thing shouldn't be merged as Draft?

Comment (michaelpj) [reply CIP-0001/README.md:144]: So presumably a CIP PR should set the status to `Proposed` and the fact of it being a PR means it's a draft?

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:105]: What about CIPs that are high quality and get merged but have nobody working towards them? Or is the idea that a CIP should not be accepted unless there _is_ a clear path to active?

File: CIP-0001/README.md
Comment (michaelpj) [inline CIP-0001/README.md:107]: What about CIPs where e.g. the implementation gets rejected? "Obsolete" doesn't feel like the right word to to cover all of "superseded", "retracted", "rejected", etc.

**Comment by michaelpj** [comment]
This is way better!

**Comment by michaelpj** [comment]
> People (including me, some time ago) wonder about cips.cardano.org - how it's different from GitHub, how it's generated.

Perhaps this repository needs a CONTRIBUTING: obviously for contributing CIPs themselves, it should link to CIP-001, but there are clearly a few other things to say, like how the website works and so on.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (michaelpj) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: On reflection, I'm a bit confused by this. This will get people to put the content of the proposed CIP _in the PR description_, rather than in a file in the PR. Is that what you want?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (michaelpj) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Should the abstract describe just the technical issue being addressed, the proposed solution, or both? I think probably both.

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:25]: > Acceptance by the CIP editors does not indicate any judgement on the technical merit of a proposal.
>
> For this reason, a CIP being merged into the repository as Draft or Proposed means nothing other than that it is procedurally sound.

Actually, a CIP merged into the repository should also be _technically sound_; but because the editors can't technically review every CIP this is mostly a task that fall to the community. Editors can mediate the conversation and should ultimately have the last word based on what the various commenters said; but this is part of the goal.

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:104]: "optional", as we won't typically merge anything as draft (we've done so in the past, but we're steadily updating them as "proposed" or "active" depending on the case).

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:144]: What you're describing is the process as outlined in CIP-0001 **currently**; but it does not reflect what is actually happening. In practice, we tend to not merge CIP as draft (anymore), but keep on iterating until they're good enough to be merged as _proposed_. There are some exceptions of course, but generally speaking, the drafting stage happens in the PR. 

This is something we want to change in the long-run of course (and use for example github discussions instead, which have threads) and I don't particularly like it. But as the title's of this PR indicates, this is meant to update CIP-0001 to reflect reality.

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:105]: The latter. But, it isn't impossible to have a path to active but not being actively worked on. The path to active should simply set the "next steps" for a CIP. Doesn't mean they need to happen immediately.

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:107]: There used to be a status "rejected" which I dropped here in favor of only Obsolete. That's perhaps not a wise decision.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (KtorZ) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Hmmm. Good point.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (KtorZ) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: Both indeed.

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:25]: Perhaps this is also an issue for CIP-001: it's not clear to me what the criteria _are_ for a CIP being accepted (also "accepted" isn't defined, I have been taking it to mean "merged into the repository").  

I think the risk with saying that a CIP must be "technically sound" to be accepted is... by whose lights? Plenty of CIPs have been merged that, while reasonably coherent, were not IMO sound. Perhaps if the CIP editors were really forceful about ensuring that CIPs get expert reviews and didn't get merged until they were acceptable, that would help. But as it is I think many merged CIPs acquire a veneer of technical soundness which isn't necessarily justified.

So I guess my thought was to describe the CIP process as explicitly conveying _less_ technical judgement, to allow well-written proposals to be merged even if they haven't been assessed. There's not really a clear way to justifiably say that a proposal is sound, so maybe we just shouldn't try :shrug:

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:104]: As a stupid person reading this table this is all very confusing to me. Can a CIP have draft status or not? Can I write it in the "status" field or not? what does it mean?

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:144]: > This is something we want to change in the long-run of course

Why? What's so bad about the drafting process corresponding to the PR stage? That's a process that people are very familiar with, whereas the idea that something gets merged while still a draft is more unusual. I certainly found it confusing.

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:105]: Then maybe you should say "'Proposed' generally means that the CIP has a clear 'Path to Active' that is being actively pursued by some entity"?

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:107]: Maybe just "inactive"?

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:25]: (points noted regarding the definition of the statuses, I will refine that).

Regarding technical soundness of CIPs, this is something we strive for and part of the reason why some CIPs take so long to move forward. We need reviews from technical folks especially on areas where editors sole knowledge do not suffice. 

Note also that some CIPs do not even have an explicit ownership when it comes to who should really approve (that's the case of pretty much any standard sitting atop of Cardano). Some others like Plutus related changes can be delegated to the Plutus maintainers for ultimate approval. 

In some cases, designated experts aren't available to share their feedback (e.g. CIP-0050 or CIP-0007). And, while as editors we do our best to get their attention, we can't force people to review. Thus, putting some proposal in a weird state where they are 'sound to the best of our knowledge'.

There's also the case of few CIPs like Cip-0025 which were written after the fact, to capture an ongoing practice. It is hard to argue with those ones as they're usually simply documenting an existing process. Perhaps the mistake was to simply accept them because of that, instead of pushing back and commenting on their shortcomings. Yet, I think it's still preferable to keep such standards in one place, even if not ideal, because this is also where we can iterate on them and come up with new approaches to replace them.

File: CIP-0001/README.md
Comment (KtorZ) [reply CIP-0001/README.md:144]: Discussions on pull requests are hard to manage and keep track of. Especially on long running CIPs with hundreds of comments. There tends to be cross chats, and as the underlying proposal gets updated, some comments become obsolete making it even harder to follow.

Although, that's also 'the best we have' for now. So maybe all that is needed is some more discipline / practices in the PR / Comment management. I don't know. Nothing set in stone yet, but I can see that there's something that _could_ be improved.

File: CIP-0001/README.md
Comment (michaelpj) [reply CIP-0001/README.md:25]: > Thus, putting some proposal in a weird state where they are 'sound to the best of our knowledge'.

I think this is the crux of it. I'm suggesting that, since checking for technical soundness is not really something the CIP process can assure in general, it should explicitly disavow any such stamp of soundness. Of _course_ getting technical review and improving the process is highly desirable, but for any random CIP there cannot be a guarantee that it meets any particular technical bar. And that's okay!

**Comment by KtorZ** [comment]
Yes, Michael did it on purpose to force the re-introduction of a "rejected" status. Smart move.

**Comment by michaelpj** [comment]
Honestly I just forgot you were going to remove it 😆

**Comment by michaelpj** [comment]
Had a little chat about the "rejected" status wrt https://github.com/cardano-foundation/CIPs/pull/336. I think that "rejected" is not quite right, since it's not the case that the CIP process has rejected this proposal. What's more accurate is to say that it is no longer supported by its author. So something like "withdrawn" is maybe more appropriate. 

However, if you want to have a single "dead" status my vote still goes to "inactive", perhaps with the suggestion that an inactive proposal should have a short note explaining why it is inactive? "Inactive: no activity", "Inactive: withdrawn by proposer", "Inactive: superseded by X"?

**Comment by KtorZ** [comment]
Looking back at how we historically handled things: I don't like inactive, nor do I like rejected (for the reason you mention). There has been only a couple of cases where we had a desire from an author to merge an "anti-CIP"; or said differently, to merge a purposely bad idea to keep a track history of discussions that happened. CIPs that are rejected are typically closed without making it to the repository. Once again, I am not saying that this is the process we want in the long-run, but it is the process we follow _de facto_. 

Since the action of proposing a purposely wrong CIP reflects an actual author's will, I think the status should also. Something that expresses "contemplated but rejected". Maybe: _discarded_ or _dropped_ ?

**Comment by michaelpj** [comment]
I think there could be other workflows. Consider e.g. a pretty good CIP that gets merged, has lots of discussion, but then eventually it turns out that for some obscure incentives reason it can't be implemented in current Cardano. We probably don't want to delete it, but it would be nice to mark it in some way that makes it clear that it's not a live proposal any more.

**Comment by rphair** [comment]
I would suggest "Invalid" (or "Declined") if not for the implication that people should be prevented from doing things differently than we've documented in CIPs.

"Disavowed" may be closer but sounds overly dramatic, like from a spy movie... as does "Sanctioned" which also has the same problem as "Invalid".

"Challenged" could incorporate all the possibilities above why a CIP "didn't make it"... but doesn't answer the next question in the mind of the reader: _Why_ was it challenged?

So, since none of these words will ever really tell the whole story, no matter how specific one or several choices would be, I also think the only option remaining is to add a qualifying phrase to `Inactive` as suggested in https://github.com/cardano-foundation/CIPs/pull/331#issuecomment-1259323553.  It shouldn't be that hard to establish (starting with the 2 or 3 cases we've seen so far) a canonical set of choices to add after `Inactive:`.

**Comment by rphair** [comment]
As far as I know, I wasn't given a change to comment about the reason for my last commit... which was to keep this fresh after the omnibus update to the top level `README.md` of https://github.com/cardano-foundation/CIPs/pull/338.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (KtorZ) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Moved to a separate file, mentioned in the body of CIP-0001

**Comment by KtorZ** [comment]
Opted for the following statuses rework:

Status   | Description
---      | ---
Draft    | An implicit status given to newly proposed CIPs that haven't yet been validated or reviewed.
Proposed | Assigned to CIP that has been reviewed, accepted and now working towards acceptance. 'Proposed' generally means that implementation is ongoing or that adoption from various actors is missing before moving to 'Active'.
Active   | The proposal has completed all steps needed for its activation. Said differently, it means observable metrics showing its adoption in the ecosystem.
Inactive | The CIP is no longer, or has never been, active. When marked 'Inactive', the status should also indicate a reason (e.g. obsolete, superseded, abandoned...). For example: `Inactive: superseded by CIP-XXXX`

**Comment by rphair** [comment]
I was also going to say that not yet `Active` proposals submitted as PR's would have to be triaged to include a _Path to Active_ if they were submitted as `Proposed`... but @KtorZ I guess you are planning an new state that does not imply review but means more than the meaningless `Draft` 🤓

**Comment by KtorZ** [comment]
I've included most feedback (I hope) in this latest version. Note that I have left the discussion / rework of the statuses for the next iteration -- the one that'll define the new process. In this one, I have left the previous statuses as they were defined (and simply removed 'On Hold' which we literally never used). 

Tried to clarified the wording on the statuses thanks to @michaelpj's feedback. I hope we can get this one on review next week and merge it A.S.A.P. so we can have a proper discussion on what the CIP process should become next.

**Review by rphair — APPROVED** [review]
Fine for this round, but still:

1 - By the time the final version comes out, prospective CIP authors should be able to see the `CIP-TEMPLATE.md` file outside the PR submission process... since the process for less experienced authors to submit a CIP will begin _long_ before they are ready to submit a PR. 😎

 - Some people will be able to find it from the line "A reference template is available in `.github/CIP-TEMPLATE.md`" but I missed this myself in the last 2 readings... maybe because of the right-alignment (why is it aligned like that?)
 - Nothing will be more obvious or useful to people than finding a link to or copy of this file somewhere they can see it: in the directory listing for CIP-0001 on GitHub.

2 - If we ask CIP submitters can be asked to include a link in their PR's OP to the proposal text (`README.md`) in their branch, it will distribute the overhead we currently have for doing this... and more importantly reduce the broken links we'll have as authors rename their files & directories without updating any link we added ourselves (without their consent or perhaps even their knowledge) in the triage process.

**Comment by KtorZ** [comment]
@rphair I think that a PR template that 

(a) reminds of CIP-0001 
(b) points to CIP-TEMPLATE.md 
(c) encourages people to also provide a rendered version of their CIP (using Github's existing capabilities) 

should address your concerns right? I did remove the previously PR template on the basis that indeed, we do not want people to generally put the CIP's content in their PR body and thus, having the CIP template defined as a PR template felt off. Or are you suggesting that we should encourage people to do exactly as such? (reading your last comment, I am not sure anymore). 

I can also add a link to the CIP-TEMPLATE.md somewhere on the root README, to make it more apparent.

**Comment by rphair** [comment]
thanks @KtorZ, the content & proposed usage of the template are fine... I'm only suggesting that this PR template _itself_ should be something that people can read and find easily, regardless of how it's used or not used in the PR submission process.

i.e. showing it _somehow_ in this directory in which the contents of the `.github` directory are suppressed: https://github.com/cardano-foundation/CIPs/tree/cip-0001-rework/CIP-0001

I do agree, and have never disputed, that stuffing the template into a PR form would encourage the CIP going into the OP which has always been messy. 😎

**Comment by rphair** [comment]
p.s., hopefully explaining myself better this time 😖 ... I'm happy with & approving this draft either way, yet also thinking from the point of view of those who only want to _learn_ (and possibly write) about the CIP process without actually writing a CIP.  It will help these visitors to CIP-0001 to see the structural template in the directory listing, and not have to look for & possibly miss it in a document reference that currently looks like a footnote. 🤔

**Comment by KtorZ** [comment]
Merging this one now, to make room for the _actual rework_ and CIP-9999.

**Pull request merged**
